lms/services/h_api.py

```diff
@@ -81,17 +80,15 @@ def _api_request(self, method, path, body=None, headers=None):
             request_args["data"] = body
 
         try:
-            response = requests.request(
+            response = self._http_service.request(
                 method=method,
                 url=self._base_url + path.lstrip("/"),
                 auth=self._http_auth,
                 timeout=10,
                 headers=headers,
                 **request_args,
             )
-            response.raise_for_status()
-
-        except RequestException as err:
+        except HTTPError as err:
```

Comment:
:man_shrugging: Not much to see here, the new `HTTPService`'s API is very similar to the `requests` API that `HAPI` was already using. This does mean that `HAPI` has picked up some `HTTPService` features that it was lacking before, e.g. using a requests session.